src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java

```diff
@@ -43,18 +47,21 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
 
         SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
 
-        // Due to a bug with CCD related to hidden fields, this field not being set
-        // on the final submission from CCD, so we need to reset it here
-        // See https://tools.hmcts.net/jira/browse/RDM-8200
-        // This is a temporary workaround for this issue.
-        sscsCaseData.setAdjournCaseGeneratedDate(LocalDate.now().toString());
-
         PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
 
         previewDocumentService.writePreviewDocumentToSscsDocument(sscsCaseData, DRAFT_ADJOURNMENT_NOTICE, sscsCaseData.getAdjournCasePreviewDocument());
 
-        if (sscsCaseData.getAdjournCaseInterpreterRequired() != null) {
-            HearingOptions hearingOptions = sscsCaseData.getAppeal().getHearingOptions() != null ? sscsCaseData.getAppeal().getHearingOptions() : HearingOptions.builder().build();
+        if (SscsUtil.isSAndLCase(sscsCaseData)
+            && isAdjournmentEnabled // TODO SSCS-10951
+            && (sscsCaseData.isAdjournCaseAbleToBeListedRightAway()
+            || isNoOrNull(sscsCaseData.getAdjournCaseAreDirectionsBeingMadeToParties()))
+        ) {
```

Comment:
Minor - Formatting for readability, you could start the line with square bracket and put the ) at end of previous line.

Reply:
Ah I see

Reply:
) { on a new line is checkstyle enforced so will keep as is